src/Symfony/Component/Translation/Bridge/Crowdin/Provider/CrowdinProvider.php

```diff
@@ -0,0 +1,395 @@
+            }
+
+            foreach ($locales as $locale) {
+                if ($locale !== $this->defaultLocale) {
+                    $content = $this->exportProjectTranslations($locale, $fileId);
+                } else {
+                    $content = $this->downloadSourceFile($fileId);
+                }
+
+                $translatorBag->addCatalogue($this->loader->load($content, $locale, $domain));
+            }
+        }
+
+        return $translatorBag;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(TranslatorBag $translatorBag): void
+    {
+        foreach($translatorBag->getDomains() as $domain) {
+            $fileId = $this->getFileId($domain);
+
+            if ($fileId) {
+                $this->deleteFile($fileId);
+            }
+        }
+
+        $this->files = [];
+    }
+
+    protected function getDefaultHost(): string
+    {
+        if ($this->organizationDomain) {
+            return sprintf('%s.%s', $this->organizationDomain, self::HOST);
+        } else {
+            return self::HOST;
+        }
+    }
+
+    protected function getDefaultHeaders(): array
+    {
+        return [
+            'Authorization' => 'Bearer ' . $this->apiToken
+        ];
+    }
+
+    private function getFileId(string $domain): int
+    {
+        if (isset($this->files[$domain]) && $this->files[$domain]) {
+            return $this->files[$domain];
+        }
+
+        $files = $this->getFilesList();
+
+        foreach($files as $file) {
+            if ($file['data']['name'] === sprintf('%s.%s', $domain, 'xlf')) {
+                $this->files[$domain] = (int)$file['data']['id'];
+
+                return $this->files[$domain];
+            }
+        }
+
+        return 0;
+    }
+
+    private function addFile(string $domain, string $content): void
+    {
+        $storageId = $this->addStorage($domain, $content);
+
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.files.getMany (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.files.getMany (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%s/files', $this->getEndpoint(), $this->projectId), [
+            'headers' => array_merge($this->getDefaultHeaders(), [
+                'Content-Type' => 'application/json',
+            ]),
+            'body' => json_encode([
+                'storageId' => $storageId,
+                'name' => sprintf('%s.%s', $domain, 'xlf'),
+            ]),
+        ]);
+
+        $responseContent = $response->getContent(false);
+
+        if (Response::HTTP_CREATED !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to add a File in Crowdin for domain "%s": "%s".', $domain, $responseContent), $response);
+        }
+
+        dump(sprintf('[Crowdin info] File "%s.%s" added successfully.', $domain, 'xlf'));
```

Comment:
`dump` is a dev only function that should be used for debug purpose only. We don't need to "display" anything here. That's the role of the Commands. Please remove all instances of `dump` call